src/anim/controller/anim-blend-tree.js

```diff
@@ -143,25 +149,26 @@ class AnimBlendTree extends AnimNode {
                 parameterValues = this._parameters.map(function (param) {
                     return this._findParameter(param).value;
                 }.bind(this));
-                if (this._parameterValues && (JSON.stringify(this._parameterValues) === JSON.stringify(parameterValues))) {
-                    return;
-                }
+                if (this.parametersEqual(parameterValues)) return;
                 this._parameterValues = parameterValues;
                 p = new Vec2(this._parameterValues);
+                var pLength = p.length();
 
 
                 weightSum = 0.0;
                 for (i = 0; i < this._children.length; i++) {
-                    pi = this._children[i].point.clone();
+                    pi = this._children[i].point;
+                    var piLength = this._children[i].pointLength;
                     minj = Number.MAX_VALUE;
                     for (j = 0; j < this._children.length; j++) {
                         if (i === j) continue;
-                        pj = this._children[j].point.clone();
+                        pj = this._children[j].point;
+                        var pjLength = this._children[j].pointLength;
                         var pipAngle = getAngleRad(pi, p);
                         var pipjAngle = getAngleRad(pi, pj);
-                        pipj = new Vec2((pj.length() - pi.length()) / ((pj.length() + pi.length()) / 2), pipjAngle * 2.0);
-                        pip = new Vec2((p.length() - pi.length()) / ((pj.length() + pi.length()) / 2), pipAngle * 2.0);
-                        result = clamp(1.0 - Math.abs((pip.clone().dot(pipj) / pipj.lengthSq())), 0.0, 1.0);
+                        pipj = new Vec2((pjLength - piLength) / ((pjLength + piLength) / 2), pipjAngle * 2.0);
+                        pip = new Vec2((pLength - piLength) / ((pjLength + piLength) / 2), pipAngle * 2.0);
```

Comment:
I'm not saying you have to fix this now, but you really want to get to the point where your code aggressively minimizes dynamic allocations. These two `Vec2`s could be preallocated and use `Vec2#set` instead.

Reply:
I've refactored the BlendTree class again which removes the creation of new Vec2's here.